packages/tanstack-query/src/procedure-utils.ts

```diff
@@ -24,46 +26,92 @@ export interface ProcedureUtils<TClientContext extends ClientContext, TInput, TO
   /**
    * Calling corresponding procedure client
    *
-   * @see {@link https://orpc.unnoq.com/docs/integrations/tanstack-query-old/basic#calling-procedure-clients Tanstack Calling Procedure Client Docs}
+   * @see {@link https://orpc.unnoq.com/docs/integrations/tanstack-query#calling-clients Tanstack Calling Procedure Client Docs}
    */
   call: Client<TClientContext, TInput, TOutput, TError>
 
+  /**
+   * Generate a **full matching** key for [Query Options](https://orpc.unnoq.com/docs/integrations/tanstack-query#query-options).
+   *
+   * @see {@link https://orpc.unnoq.com/docs/integrations/tanstack-query#query-mutation-key Tanstack Query/Mutation Key Docs}
+   */
+  queryKey(
+    ...rest: MaybeOptionalOptions<
+      QueryKeyOptions<TInput>
+    >
+  ): DataTag<QueryKey, TOutput, TError>
+
   /**
    * Generate options used for useQuery/useSuspenseQuery/prefetchQuery/...
    *
-   * @see {@link https://orpc.unnoq.com/docs/integrations/tanstack-query-old/basic#query-options-utility Tanstack Query Options Utility Docs}
+   * @see {@link https://orpc.unnoq.com/docs/integrations/tanstack-query#query-options Tanstack Query Options Utility Docs}
    */
   queryOptions<U, USelectData = TOutput>(
     ...rest: MaybeOptionalOptions<
       U & QueryOptionsIn<TClientContext, TInput, TOutput, TError, USelectData>
     >
   ): NoInfer<U & Omit<QueryOptionsBase<TOutput, TError>, keyof U>>
```

Comment:
Would be really nice if this were to also return the queryOptions with a datatag as queryKey. When working with optimistic updates from mutations, I like defining the queryOptions once, and then just reuse those throughout the onSubmit, onSuccess, etc. handlers to update the query cache for example. By adding the datatag to the return value here, the queryOptions.queryKey automatically is typed. Right now I'm wrapping my orpc.some.procedure.queryOptions() with the queryOptions util from tanstack query, just to get the typed queryKey.